### PR TITLE
Multiple beam handling

### PR DESCRIPTION
**Pull request by keflavich**
WIP.  Solves https://github.com/radio-astro-tools/spectral-cube/issues/436

Goal is to avoid trying to do anything (such as convolve) channels that have had their beams flagged out.

### CODE REVIEW

**Comment by keflavich** [comment]
Relies on https://github.com/radio-astro-tools/radio-beam/pull/51

File: spectral_cube/spectral_cube.py [hunk 1]
```diff
@@ -3300,3 +3360,36 @@ def to(self, unit, equivalencies=()):
...
+        Helper function to mask out channels.  This function is equivalent to
+        adding a mask with ``cube[view]`` where ``view`` is broadcastable to
+        the cube shape, but it accepts 1D arrays that are not normally
+        broadcastable.  Additionally, for `VaryingResolutionSpectralCube`s,
+        the beams in the bad channels will not be checked when averaging,
+        convolving, and doing other operations that are multibeam-aware.
+
+        Parameters
+        ----------
+        badchannels : array
```
Comment (e-koch) [inline spectral_cube/spectral_cube.py]: `goodchannels`

Comment (e-koch) [reply spectral_cube/spectral_cube.py]: Maybe be explicit that this is a 1D array, too

File: spectral_cube/spectral_cube.py [hunk 2]
```diff
@@ -3044,17 +3079,19 @@ def mask_out_bad_beams(self, threshold, reference_beam=None,
...
 
-        includemask = BooleanArrayMask(includemask[:,None,None],
+        includemask = BooleanArrayMask(goodbeams[:,None,None],
                                        self._wcs,
                                        shape=self._data.shape)
 
         return self._new_cube_with(mask=self.mask & includemask,
-                                   beam_threshold=threshold)
+                                   beam_threshold=threshold,
+                                   goodbeams_mask=self._goodbeams_mask & ~goodbeams,
```
[the remark this answers is not in the document]
Comment (keflavich) [reply spectral_cube/spectral_cube.py]: Yes.  I had used `badbeams` and missed this when switching.  Thanks!

**Pull request merged**